src/box/memtx_tx.c

```diff
@@ -1910,10 +1918,28 @@ memtx_tx_tuple_clarify_impl(struct txn *txn, struct space *space,
 		if (memtx_tx_story_is_visible(story, txn, is_prepared_ok,
 					      &result, &own_change))
 			break;
+		if (story->add_psn != 0 && story->add_stmt != NULL &&
+		    txn != NULL) {
+			/*
+			 * If we skip prepared story then the transaction
+			 * must be before prepared in serialization order.
+			 * That can be a read view or conflict already.
+			 */
+			memtx_tx_handle_conflict(story->add_stmt->txn, txn);
+		}
 		if (story->link[index->dense_id].older_story == NULL)
 			break;
 		story = story->link[index->dense_id].older_story;
 	}
+	if (tuple != NULL && story->del_psn != 0 && story->del_stmt != NULL &&
+	    txn != NULL) {
+		/*
+		 * If we see a tuple that is deleted by prepared transaction
+		 * then the transaction must be before prepared in serialization
+		 * order. That can be a read view or conflict already.
+		 */
+		memtx_tx_handle_conflict(story->del_stmt->txn, txn);
```

Comment:
Ditto.

Reply:
I beg your pardon: next time I won’t be using ditto, as it does mess things up — even I’m not sure what I meant here.

Guess it was related to this comment above:
> Please consider updating the documentation of `memtx_tx_handle_conflict`:
> 
> https://github.com/tarantool/tarantool/blob/f1d750f036956873c79bf5de51f77b38da25ae0f/src/box/memtx_tx.h#L215
> 
> 
> Here we are handling a conflict after preparation of transaction.